### PR TITLE
Drop direct subject reference in favour of consumer

### PR DESCRIPTION
**Pull request by mensfeld**
Replace direct `subject` usage in favour of named `consumer` and do not accumulate messages on consumer unless consumer is a karafka consumer. 

ref https://github.com/karafka/karafka-testing/issues/106

### CODE REVIEW

File: lib/karafka/testing/rspec/helpers.rb [hunk 1]
```diff
@@ -85,13 +85,15 @@ def _karafka_consumer_for(requested_topic, requested_consumer_group = nil)
         #     karafka.produce({ 'hello' => 'world' }.to_json, 'partition' => 6)
         #   end
         def _karafka_add_message_to_consumer_if_needed(message)
-          # We're interested in adding message to subject only when it is a consumer
+          # Consumer needs to be defined in order to pass messages to it
+          return unless respond_to?(:consumer)
```
Comment (ojab) [inline lib/karafka/testing/rspec/helpers.rb:89]: Maybe better to `defined?(:consumer)`, it will make local variables work.

**Review by ojab — APPROVED** [review]

File: lib/karafka/testing/rspec/helpers.rb (hunk 1 above)
Comment (mensfeld) [reply lib/karafka/testing/rspec/helpers.rb:89]: `defined?(:consumer)` always returns for me value `expression`, never false nor nil and this breaks the whole code as it always passes

Comment (ojab) [reply lib/karafka/testing/rspec/helpers.rb:89]: oh, it should be `defined?(consumer)`, not `:consumer`

Comment (mensfeld) [reply lib/karafka/testing/rspec/helpers.rb:89]: good point, works

**Pull request merged**